src/ext/components/partial-dependence-input.coffee

```diff
@@ -1,57 +1,123 @@
 H2O.PartialDependenceInput = (_, _go) ->
-  _exception = signal null #TODO display in .jade
+  _exception = signal null 
   _destinationKey = signal "ppd-#{Flow.Util.uuid()}"
 
   _frames = signals []
   _models = signals []
   _selectedModel = signals null
   _selectedFrame = signal null
+  _useCustomColumns = signal no
+  _columns = signal []
   _nbins = signal 20
 
-  # _leftColumns = signals []
-  # _selectedLeftColumn = signal null
-  # _includeAllLeftRows = signal false
 
-  # _selectedRightFrame = signal null
-  # _rightColumns = signals []
-  # _selectedRightColumn = signal null
-  # _includeAllRightRows = signal false
+  # search & filter functionalities
+  _visibleItems = signal []
+  _filteredItems = signal []
+
+  MaxItemsPerPage = 100
+
+  _currentPage = signal 0
+  _maxPages = lift _filteredItems, (entries) -> Math.ceil entries.length / MaxItemsPerPage
+  _canGoToPreviousPage = lift _currentPage, (index) -> index > 0
+  _canGoToNextPage = lift _maxPages, _currentPage, (maxPages, index) -> index < maxPages - 1
+
+  _selectionCount = signal 0
+
+  _isUpdatingSelectionCount = no
+
+  _searchTerm = signal ''
+  _searchCaption = lift _columns, _filteredItems, _selectionCount, _currentPage, _maxPages, (entries, filteredItems, selectionCount, currentPage, maxPages) ->
+    caption = if maxPages is 0 then '' else "Showing page #{currentPage + 1} of #{maxPages}."
+    if filteredItems.length isnt entries.length
+      caption += " Filtered #{filteredItems.length} of #{entries.length}."
+    if selectionCount isnt 0
+      caption += " #{selectionCount} selected for PDP calculations."
+    caption
+
+  blockSelectionUpdates = (f) ->
+    _isUpdatingSelectionCount = yes
+    f()
+    _isUpdatingSelectionCount = no
+
+  incrementSelectionCount = (amount) ->
+    _selectionCount _selectionCount() + amount
+
+  _hasFilteredItems = lift _columns, (entries) -> entries.length > 0
+
+
+  filterItems = ->
+    searchTerm = _searchTerm().trim()
+
+    filteredItems = []
+
+    for entry, i in _columns()
+      hide = no
+      if (searchTerm isnt '') and -1 is entry.value.toLowerCase().indexOf searchTerm.toLowerCase()
+        hide = yes
+
+      unless hide
+        filteredItems.push entry
+
+    _filteredItems filteredItems
+
+    start = _currentPage() * MaxItemsPerPage
+    _visibleItems _filteredItems().slice start, start + MaxItemsPerPage
+
+  # when searchterm changes, filterItems is called
+  react _searchTerm, throttle filterItems, 500
```

Comment:
this says bind the value of `_searchTerm` to the functio `filterItems` (no more than once every 500ms)